From e0a4fef68417696e756b70bfcf8e70578aada7eb Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Thu, 6 Aug 2020 15:16:06 +0800
Subject: [PATCH] ARM: dts: rv1126: add evb tb robot dts of V13 board

add the support of v13 for evb tb and robot product forms.

Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
Change-Id: I45cc935579997210805b4c5d41415dd06b468b4f
---
 arch/arm/boot/dts/Makefile                    |  2 ++
 .../boot/dts/rv1126-evb-ddr3-v13-robot.dts    | 17 ++++++++++++++
 .../boot/dts/rv1126-evb-ddr3-v13-tb-emmc.dts  | 19 ++++++++++++++++
 arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts     | 17 +-------------
 arch/arm/boot/dts/rv11xx-evb-v13.dtsi         | 22 +++++++++++++++++++
 5 files changed, 61 insertions(+), 16 deletions(-)
 create mode 100644 arch/arm/boot/dts/rv1126-evb-ddr3-v13-robot.dts
 create mode 100644 arch/arm/boot/dts/rv1126-evb-ddr3-v13-tb-emmc.dts
 create mode 100644 arch/arm/boot/dts/rv11xx-evb-v13.dtsi

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 4da2085ed052..c38182b98a8b 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -858,6 +858,8 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += \
 	rv1126-evb-ddr3-v12-tb-emmc.dtb \
 	rv1126-evb-ddr3-v12-tb-spi-nor.dtb \
 	rv1126-evb-ddr3-v13.dtb \
+	rv1126-evb-ddr3-v13-robot.dtb \
+	rv1126-evb-ddr3-v13-tb-emmc.dtb \
 	rv1126-iotest-v10.dtb \
 	rv1126-ipc2-ddr3-v10.dtb \
 	rv1126-sphericalipc-ddr3-v10.dtb \
diff --git a/arch/arm/boot/dts/rv1126-evb-ddr3-v13-robot.dts b/arch/arm/boot/dts/rv1126-evb-ddr3-v13-robot.dts
new file mode 100644
index 000000000000..1f3ebe352436
--- /dev/null
+++ b/arch/arm/boot/dts/rv1126-evb-ddr3-v13-robot.dts
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2020 Rockchip Electronics Co., Ltd.
+ */
+/dts-v1/;
+#include "rv1126.dtsi"
+#include "rv11xx-evb-v13.dtsi"
+#include "rv11xx-evb-v10-robot.dtsi"
+
+/ {
+	model = "Rockchip RV1126 EVB DDR3 V13 Board For Robot";
+	compatible = "rockchip,rv1126-evb-ddr3-v13-robot", "rockchip,rv1126";
+
+	chosen {
+		bootargs = "console=ttyFIQ0 root=PARTUUID=614e0000-0000 rootwait snd_aloop.index=7";
+	};
+};
diff --git a/arch/arm/boot/dts/rv1126-evb-ddr3-v13-tb-emmc.dts b/arch/arm/boot/dts/rv1126-evb-ddr3-v13-tb-emmc.dts
new file mode 100644
index 000000000000..53f5e8d62ddf
--- /dev/null
+++ b/arch/arm/boot/dts/rv1126-evb-ddr3-v13-tb-emmc.dts
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2020 Rockchip Electronics Co., Ltd.
+ */
+
+/dts-v1/;
+#include "rv1126.dtsi"
+#include "rv11xx-evb-v13.dtsi"
+#include "rv11xx-thunder-boot-emmc.dtsi"
+#include "rv11xx-evb-thunder-boot.dtsi"
+
+/ {
+	model = "Rockchip RV1126 EVB DDR3 V13 with eMMC (Thunder Boot)";
+	compatible = "rockchip,rv1126-evb-ddr3-v13-tb-emmc", "rockchip,rv1126";
+
+	chosen {
+		bootargs = "loglevel=0 initcall_nr_threads=-1 initcall_debug=0 printk.devkmsg=on root=/dev/rd0 console=ttyFIQ0 snd_aloop.index=7 driver_async_probe=dwmmc_rockchip rk.root2nd=/dev/mmcblk0p7";
+	};
+};
diff --git a/arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts b/arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts
index 4e2bec9abf00..5dd95a841ae3 100644
--- a/arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts
+++ b/arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts
@@ -4,7 +4,7 @@
  */
 /dts-v1/;
 #include "rv1126.dtsi"
-#include "rv11xx-evb-v12.dtsi"
+#include "rv11xx-evb-v13.dtsi"
 / {
 	model = "Rockchip RV1126 EVB DDR3 V13 Board";
 	compatible = "rockchip,rv1126-evb-ddr3-v13", "rockchip,rv1126";
@@ -12,20 +12,5 @@
 	chosen {
 		bootargs = "earlycon=uart8250,mmio32,0xff570000 console=ttyFIQ0 root=PARTUUID=614e0000-0000 rootfstype=ext4 rootwait snd_aloop.index=7";
 	};
-
-	backlight: backlight {
-		pwms = <&pwm0 0 25000 0>;
-	};
-};
-
-&u2phy0 {
-	vup-gpios = <&gpio0 RK_PC1 GPIO_ACTIVE_LOW>;
 };
 
-&pwm0 {
-	status = "okay";
-};
-
-&pwm3 {
-	status = "disabled";
-};
diff --git a/arch/arm/boot/dts/rv11xx-evb-v13.dtsi b/arch/arm/boot/dts/rv11xx-evb-v13.dtsi
new file mode 100644
index 000000000000..d4d361b92ea9
--- /dev/null
+++ b/arch/arm/boot/dts/rv11xx-evb-v13.dtsi
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2020 Rockchip Electronics Co., Ltd.
+ */
+
+#include "rv11xx-evb-v12.dtsi"
+
+&backlight {
+	pwms = <&pwm0 0 25000 0>;
+};
+
+&pwm0 {
+	status = "okay";
+};
+
+&pwm3 {
+	status = "disabled";
+};
+
+&u2phy0 {
+	vup-gpios = <&gpio0 RK_PC1 GPIO_ACTIVE_LOW>;
+};
-- 
2.35.3

